### ISSUE TITLE
Implementation of selective disambiguation

### ISSUE THREAD

**Issue by kiminh** [rktamplayo/Entity2Topic#1]
Hi,
I've read your paper "Entity Commonsense Representation for Neural Abstractive Summarization`. Good idea. However, I'm a little confused about the implementation of the selective disambiguation module in your code.

![image](https://user-images.githubusercontent.com/8201408/58149558-0e65c500-7c96-11e9-89d6-6ce9ac7f3150.png)

where `e_{i}` is the entity embedding vector, `e^{'}_{i}` is the new entity vector after disambiguating encoder (where globally or locally)

I mean `e^{'}_{i}` is `entity_vector`, where `d` in your formula is a scalar. `e_{i}` is `entity_proj `

`d = \sigma(W_d e^{'}_{i} + b_{d}` equals to 
```
                Wt = tf.get_variable("Wt", [1000, 1], initializer=xavier_initializer())
                bt = tf.Variable(tf.constant(0.0, shape=[1]))
                t = tf.nn.sigmoid(tf.tensordot(entity_vector, Wt, 1) + bt)
```

here `f` you use `tanh`, then 
`f(W_x   e_{i} + b_x)` equals to
```
                Wh = tf.get_variable("Wh", [1000, 1000], initializer=xavier_initializer())
                bh = tf.Variable(tf.constant(0.0, shape=[1000]))
                entity_proj = tf.nn.tanh(tf.tensordot(entity_proj, Wh, 1) + bh)
```

I'm doubting whether following implementation equals to  `d * f(W_x   e_{i} + b_x) + (1-d)* f(W_y   e^{'}_{i} + b_y)`
```
entity_vector = t*entity_vector + (1-t)*entity_proj
```
 **I mean the correct implementation should be** 
```
                Wy = tf.get_variable("Wy", [1000, 1000], initializer=xavier_initializer())
                by = tf.Variable(tf.constant(0.0, shape=[1000]))
                entity_vector = tf.nn.tanh(tf.tensordot(entity_vector, Wy, 1) + by)

                entity_vector = t*entity_vector + (1-t)*entity_proj
```
code snippet in `bigru_model.py`

```
            if highway:
                # y
#                entity_proj = entity_inputs_emb[:,3:-3,:]
                Wh = tf.get_variable("Wh", [1000, 1000], initializer=xavier_initializer())
                bh = tf.Variable(tf.constant(0.0, shape=[1000]))
                entity_proj = tf.nn.tanh(tf.tensordot(entity_proj, Wh, 1) + bh)
                if not forward_only:
                    entity_proj.set_shape([self.batch_size, None, 1000])
                else:
                    entity_proj.set_shape([self.batch_size*10, None, 1000])
                
                if not forward_only:
                    entity_proj = tf.nn.dropout(entity_proj, keep_prob=0.5)
                
                # t
                Wt = tf.get_variable("Wt", [1000, 1], initializer=xavier_initializer())
                bt = tf.Variable(tf.constant(0.0, shape=[1]))
                t = tf.nn.sigmoid(tf.tensordot(entity_vector, Wt, 1) + bt)
                if not forward_only:
                    t.set_shape([self.batch_size, None, 1000])
                else:
                    t.set_shape([self.batch_size*10, None, 1000])
                self.t = t
                
                entity_vector = t*entity_vector + (1-t)*entity_proj
```

**Comment by kiminh** [comment]
I mean in your implementation is a **highway network style**(use a transforming gate to decide whether use the entity vector after encoder or carry the original embedding entity vector). I think it's the true implementation of the selective disambiguation module. However, it's not consistent with the formula in your paper.

formula(3) in paper "Highway Networks"
![image](https://user-images.githubusercontent.com/8201408/58155628-d2872b80-7ca6-11e9-8f52-a53bdc846a48.png)

**Comment by rktamplayo** [comment]
Hi Hui,

I also have received your email. I'm responding here instead.

Thanks for opening this up. I also received an email before this stating the same issue. I actually have uploaded a previous version of the code, so you are correct in spotting the issue in the code.

I was supposed to reupload the code, but I totally forgot about it. I’ll upload it as soon as possible (just finished overtiming with EMNLP so probably a little bit later...)

The formula(3) in the Highway Networks paper is a little different with what we used in the NAACL paper. Everything is the same except for the fact that we do another non-linear transformation to both e_i and e'_i (x and H(x, W_h) in the original HN paper). In our prior experiments, this gives us a little boost in performance, which is why we modified the original HNs.

Reinald Kim
